### DISCUSSION TITLE
ScriptReference/MonoBehaviour.OnDrawGizmos

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#27] [Notes]
# ScriptReference/MonoBehaviour.OnDrawGizmos

https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnDrawGizmos.html

**Comment by vertxxyz** [comment]
You can also implement Gizmos using the [`DrawGizmo`](https://docs.unity3d.com/ScriptReference/DrawGizmo.html) attribute, it has a [`GizmoType`](https://docs.unity3d.com/ScriptReference/GizmoType.html) flag that's more flexible the than messages sent to the object. This allows you to move Gizmo code out of the runtime script.

**Comment by docscomments** [comment]
**Comment by haowan:**  

[Here](http://answers.unity3d.com/questions/124286/how-to-pick-ondrawgizmos-gizmo.html) is a solution for those wanting to draw their own pickable gizmos.
